active/0000-plaforms.md

```diff
@@ -0,0 +1,328 @@
+```
+
+**Test a specific platform.**
+```cli
+ember t -p "<type>[-<name>]"
+```
+
+**Deploy a specific platform.**
+```cli
+ember d <deployTarget> -p "<type>[-<name>]" 
+```
+
+This is short hand for:
+```cli
+ember deploy --deployTarget="<deployTarget>" --platform="<type>" [--target="<name>"] --environment="development"
+```
+
+
+### Platform Specific Code
+
+Platforms would leverage in-repo-addon semantics and Broccoli plugins to enable
+sprinkling of platform specific code into an app as needed with "platform addons".
+
+A platform addon is generated automatically for each platform created with `ember platform`.
+
+```
+<project>
+  /lib
+    /<type>[-<name>]
+```
+
+These "addons" function differently from normal addons, and would be prefixed with `-platform-`.
+Instead of code in the app overriding or importing code from the addon, the addon would 
+take precedence.  This let's you selectively replace or add entire modules as needed.
+
+You can import the corresponding `app` version of a module into the platform addon.
+
+```js
+import Foo from '<project-name>/routes/foo';
+
+// our platform version of routes/foo
+export default Foo.extend({
+  hello: 'World!'
+});
+```
+
+### Platform Specific Config
+
+A platform config file is generated automatically for each platform you create with `ember platform`.
+```
+<project>
+  /config
+    /platforms
+      /<type>[-<name>].js
+```
+
+This configuration will be merged with your primary configuration.
+
+## Plugins
+
+Users should be able to create platform plugins as addons which implement 
+any number of specified hooks. These plugins should carry in the same tradition
+in form, function and hook names as `ember-cli-deploy`.
+
+## Creating Plugins
+
+Platforms have several forms of plugins:
+
+1. `process hooks` which add behaviors to one or more platforms.
+2. `platform plugins` which implement support and generators for a specific platform.
+3. `deploy plugins` which implement support for deployment of a platform.
+
+Platform plugins (generally) are also process hooks.
+
+## All plugins
+
+Platform plugins are ember-cli addons with a few platform specific traits:
+
+1. they contain a package.json keyword to identify them as plugins
+2. they are named ember-cli-platform-*
+3. they return an object that implements one or more of the ember-cli-platforms pipeline hooks
+
+
+### Creating a Process Hook
+
+#### Create an addon
+
+```cli
+ember addon ember-cli-platform-<name>
+```
+
+#### Mark the addon as a plugin
+
+Identify your addon as a plugin by updating the package.json like so:
+
+```js
+// package.json
+
+"keywords": [
+  "ember-addon",
+
+  // if present, indicates this addon contains process hooks for ember-cli-platforms
+  "ember-cli-platform-plugin",
+  
+  // if present, indicates this addon contains a platform implementation for ember-cli-platforms
+  "ember-cli-platform",
+  
+  // if present, indicates this addon implements deployment hooks
+  "ember-cli-platform-deploy"
```

Comment:
We can get by with one.

Ostensibly, this is for permissions elevation.  In the original conception, `ember-cli-platform` meant an addon worked with the `ember platform` command, `ember-cli-platform-plugin` meant it would work with `build`, `serve` and `test` commands, and `ember-cli-platform-deploy` meant it would work with the `deploy` command.

Keeping this distinction probably would let us be a bit faster at booting the commands, but it's definitely not essential.